test/pylib/resource_gather.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+import getpass
+import logging
+import os
+import platform
+import subprocess
+from abc import ABC
+from datetime import datetime
+from pathlib import Path
+from typing import TextIO
+
+import psutil
+from attrs import define, field, asdict
+
+from test.pylib.db.model import TestMetricRecord, ResourceTimelineRecord
+from test.pylib.db.writer import DATE_TIME_TEMPLATE, SQLiteWriter, RESOURCE_TIMELINE_TABLE, DB_PATH, METRICS_TABLE
+
+CGROUP_INITIAL: Path = Path('/sys/fs/cgroup/initial')
+CGROUP_TESTS: Path = CGROUP_INITIAL.parent / 'tests'
+
+
+@define
```

Comment:
can't see the advantage of using @define here. 

regular def __init__ will be more readable for me